src/cargo/ops/cargo_compile.rs

```diff
@@ -182,7 +183,7 @@ pub fn compile_pkg<'a>(root_package: &Package,
         Some(args) => {
             if to_builds.len() == 1 {
                 let targets = try!(generate_targets(to_builds[0], profiles,
-                                                    mode, filter, release));
+                                                    mode, filter, all_feature, release));
```

Comment:
Like below, it may be best to load the `all_feature` variable just above this using `to_builds[0]` to ensure it's a contained borrow.